{"id": "CVE-2020-14422", "sourceIdentifier": "cve@mitre.org", "published": "2020-06-18T14:15:11.047", "lastModified": "2023-11-07T03:17:14.010", "vulnStatus": "Modified", "cveTags": [], "descriptions": [{"lang": "en", "value": "Lib/ipaddress.py in Python through 3.8.3 improperly computes hash values in the IPv4Interface and IPv6Interface classes, which might allow a remote attacker to cause a denial of service if an application is affected by the performance of a dictionary containing IPv4Interface or IPv6Interface objects, and this attacker can cause many dictionary entries to be created. This is fixed in: v3.5.10, v3.5.10rc1; v3.6.12; v3.7.9; v3.8.4, v3.8.4rc1, v3.8.5, v3.8.6, v3.8.6rc1; v3.9.0, v3.9.0b4, v3.9.0b5, v3.9.0rc1, v3.9.0rc2."}, {"lang": "es", "value": "La biblioteca Lib/ipaddress.py en Python versiones hasta 3.8.3, calcula inapropiadamente los valores de hash en las clases IPv4Interface e IPv6Interface, lo que podr\u00eda permitir a un atacante remoto causar una denegaci\u00f3n de servicio si una aplicaci\u00f3n est\u00e1 afectada por el desempe\u00f1o de un diccionario que contiene objetos de IPv4Interface o IPv6Interface, y este atacante puede causar que muchas entradas de diccionario sean creadas. Esto esta corregido en las versiones: v3.5.10, v3.5.10rc1; v3.6.12; v3.7.9; v3.8.4, v3.8.4rc1, v3.8.5, v3.8.6, v3.8.6rc1; v3.9.0, v3.9.0b4, v3.9.0b5, v3.9.0rc1, v3.9.0rc2"}], "metrics": {"cvssMetricV31": [{"source": "nvd@nist.gov", "type": "Primary", "cvssData": {"version": "3.1", "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H", "attackVector": "NETWORK", "attackComplexity": "HIGH", "privilegesRequired": "NONE", "userInteraction": "NONE", "scope": "UNCHANGED", "confidentialityImpact": "NONE", "integrityImpact": "NONE", "availabilityImpact": "HIGH", "baseScore": 5.9, "baseSeverity": "MEDIUM"}, "exploitabilityScore": 2.2, "impactScore": 3.6}], "cvssMetricV2": [{"source": "nvd@nist.gov", "type": "Primary", "cvssData": {"version": "2.0", "vectorString": "AV:N/AC:M/Au:N/C:N/I:N/A:P", "accessVector": "NETWORK", "accessComplexity": "MEDIUM", "authentication": "NONE", "confidentialityImpact": "NONE", "integrityImpact": "NONE", "availabilityImpact": "PARTIAL", "baseScore": 4.3}, "baseSeverity": "MEDIUM", "exploitabilityScore": 8.6, "impactScore": 2.9, "acInsufInfo": false, "obtainAllPrivilege": false, "obtainUserPrivilege": false, "obtainOtherPrivilege": false, "userInteractionRequired": false}]}, "weaknesses": [{"source": "nvd@nist.gov", "type": "Primary", "description": [{"lang": "en", "value": "CWE-330"}, {"lang": "en", "value": "CWE-682"}]}], "configurations": [{"nodes": [{"operator": "OR", "negate": false, "cpeMatch": [{"vulnerable": true, "criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*", "versionStartIncluding": "3.0.0", "versionEndExcluding": "3.5.10", "matchCriteriaId": "0DCA021D-A322-4442-A030-B723D8F8C7E5"}, {"vulnerable": true, "criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*", "versionStartIncluding": "3.6.0", "versionEndExcluding": "3.6.12", "matchCriteriaId": "7EB6A6FC-6C82-4543-8DAA-96FF2BACDB5C"}, {"vulnerable": true, "criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*", "versionStartIncluding": "3.7.0", "versionEndExcluding": "3.7.9", "matchCriteriaId": "045171F0-A76D-4385-B2AE-51479B425C45"}, {"vulnerable": true, "criteria": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*", "versionStartIncluding": "3.8.0", "versionEndExcluding": "3.8.4", "matchCriteriaId": "E9409BD5-670F-4C69-A4D2-ACF9AD8F335B"}, {"vulnerable": true, "criteria": "cpe:2.3:o:opensuse:leap:15.1:*:*:*:*:*:*:*", "matchCriteriaId": "B620311B-34A3-48A6-82DF-6F078D7A4493"}, {"vulnerable": true, "criteria": "cpe:2.3:o:opensuse:leap:15.2:*:*:*:*:*:*:*", "matchCriteriaId": "B009C22E-30A4-4288-BCF6-C3E81DEAF45A"}]}]}, {"nodes": [{"operator": "OR", "negate": false, "cpeMatch": [{"vulnerable": true, "criteria": "cpe:2.3:o:fedoraproject:fedora:31:*:*:*:*:*:*:*", "matchCriteriaId": "80F0FA5D-8D3B-4C0E-81E2-87998286AF33"}, {"vulnerable": true, "criteria": "cpe:2.3:o:fedoraproject:fedora:32:*:*:*:*:*:*:*", "matchCriteriaId": "36D96259-24BD-44E2-96D9-78CE1D41F956"}]}]}, {"nodes": [{"operator": "OR", "negate": false, "cpeMatch": [{"vulnerable": true, "criteria": "cpe:2.3:a:oracle:enterprise_manager_ops_center:12.4.0.0:*:*:*:*:*:*:*", "matchCriteriaId": "B095CC03-7077-4A58-AB25-CC5380CDCE5A"}]}]}], "references": [{"url": "http://lists.opensuse.org/opensuse-security-announce/2020-07/msg00003.html", "source": "cve@mitre.org", "tags": ["Mailing List", "Third Party Advisory"]}, {"url": "http://lists.opensuse.org/opensuse-security-announce/2020-07/msg00006.html", "source": "cve@mitre.org", "tags": ["Mailing List", "Third Party Advisory"]}, {"url": "http://lists.opensuse.org/opensuse-security-announce/2020-07/msg00032.html", "source": "cve@mitre.org", "tags": ["Mailing List", "Third Party Advisory"]}, {"url": "http://lists.opensuse.org/opensuse-security-announce/2020-07/msg00041.html", "source": "cve@mitre.org", "tags": ["Mailing List", "Third Party Advisory"]}, {"url": "https://bugs.python.org/issue41004", "source": "cve@mitre.org", "tags": ["Vendor Advisory"]}, {"url": "https://github.com/python/cpython/pull/20956", "source": "cve@mitre.org", "tags": ["Issue Tracking", "Third Party Advisory"]}, {"url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00011.html", "source": "cve@mitre.org", "tags": ["Third Party Advisory"]}, {"url": "https://lists.debian.org/debian-lts-announce/2023/05/msg00016.html", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/36XI3EEQNMHGOZEI63Y7UV6XZRELYEAU/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CNHPQGSP2YM3JAUD2VAMPXTIUQTZ2M2U/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CTUNTBJ3POHONQOTLEZC46POCIYYTAKZ/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/FCCZTAYZATTNSNEAXWA7U3HCO2OVQKT5/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/LE4O3PNDNNOMSKHNUKZKD3NGHIFUFDPX/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/NTBKKOLFFNHG6CM4ACDX4APHSD5ZX5N4/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/OXI72HIHMXCQFWTULUXDG7VDA2BCYL4Y/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/V3TALOUBYU2MQD4BPLRTDQUMBKGCAXUA/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/V53P2YOLEQH4J7S5QHXMKMZYFTVVMTMO/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/VT4AF72TJ2XNIKCR4WEBR7URBJJ4YZRD/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/X36Y523UAZY5QFXZAAORNFY63HLBWX7N/", "source": "cve@mitre.org"}, {"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YILCHHTNLH4GG4GSQBX2MZRKZBXOLCKE/", "source": "cve@mitre.org"}, {"url": "https://security.gentoo.org/glsa/202008-01", "source": "cve@mitre.org", "tags": ["Third Party Advisory"]}, {"url": "https://security.netapp.com/advisory/ntap-20200724-0004/", "source": "cve@mitre.org", "tags": ["Third Party Advisory"]}, {"url": "https://usn.ubuntu.com/4428-1/", "source": "cve@mitre.org", "tags": ["Third Party Advisory"]}, {"url": "https://www.oracle.com/security-alerts/cpujan2021.html", "source": "cve@mitre.org", "tags": ["Third Party Advisory"]}], "techniques": [{"techniqueID": "TA0003", "comment": "CVE-2020-14422 is a CPU vulnerability in Python's ipaddress module, allowing attackers to cause a denial-of-service by inducing a high CPU load. This can be mitigated by updating to affected versions of Python (3.5.10, 3.6.12, 3.7.9, 3.8.4) and applying careful server management practices to prevent unintended consequences."}], "stix_bundle": {"spec_version": "2.1", "id": "https://example.com/cve-2020-14422", "type": "bundle", "objects": [{"id": "https://example.com/cve-2020-14422", "type": "vulnerability", "name": "CVE-2020-14422", "description": "Affected Python versions (3.5.10, 3.6.12, 3.7.9, 3.8.4) have a high CPU load caused by induced dictionary attacks, leading to a denial-of-service.", "references": [{"id": "https://example.com/cve-2020-14422-reference-1", "type": "reference", "name": "CVE-2020-14422 Reference 1", "description": "This is a reference to the first reference."}, {"id": "https://example.com/cve-2020-14422-reference-2", "type": "reference", "name": "CVE-2020-14422 Reference 2", "description": "This is a reference to the second reference."}], "threats": [{"id": "https://example.com/cve-2020-14422-threat-1", "type": "threat", "name": "Unpatched Python Vulnerability Threat", "description": "Attackers can exploit the unpatched Python vulnerability to cause a denial-of-service."}]}, {"id": "https://example.com/ta0003", "type": "technique", "name": "TA0003", "description": "This technique is used to describe the actions taken by the attacker.", "references": [{"id": "https://example.com/ta0003-reference-1", "type": "reference", "name": "TA0003 Reference 1", "description": "This is a reference to the first reference."}, {"id": "https://example.com/ta0003-reference-2", "type": "reference", "name": "TA0003 Reference 2", "description": "This is a reference to the second reference."}], "mitigations": [{"id": "https://example.com/cve-2020-14422-mitigation-1", "type": "mitigation", "name": "Update to affected Python versions", "description": "Updating to affected Python versions can mitigate the vulnerability."}, {"id": "https://example.com/cve-2020-14422-mitigation-2", "type": "mitigation", "name": "Apply careful server management practices", "description": "Applying careful server management practices can help prevent unintended consequences."}]}]}, "technical_analysis": {"Introduction": "Introduction content.", "Impact and Scope": "Details on the impact and scope of the CVE, mention cvss score with version and attack vector.", "Related MITRE ATT&CK Techniques": "List of related MITRE ATT&CK Techniques and their relevance to the CVE.", "Technical Details": "In-depth technical details of the CVE.", "Detection": "Methods for detecting the CVE.", "Mitigation": "Mitigation strategies for the CVE.", "Conclusion": "Summary and conclusions."}, "executive_analysis": {"Overview": "Overview of the CVE.", "Business Impact": "Explanation of the business impact of the CVE.", "Technical Impact": "Explanation of the technical impact of the CVE.", "Mitigation Strategies": "High-level mitigation strategies.", "Recommendations": "Recommendations for executives.", "Conclusions": "Final conclusions and next steps."}}